src/NuGet.Clients/NuGet.SolutionRestoreManager/VSNominationUtilities.cs

```diff
@@ -510,10 +514,10 @@ private static ProjectRestoreReference ToProjectRestoreReference(IVsReferenceIte
             return dependency;
         }
 
-        private static bool TryGetVersionRange(IVsReferenceItem item, string propertyName, out VersionRange versionRange)
+        private static bool TryGetVersionRange(IVsReferenceItem2 item, string propertyName, [NotNullWhen(true)] out VersionRange? versionRange)
```

Comment:
Looking at this method, I think it'd throw if the provided version range is unparsable right? 

It makes the "try" name confusing.

Reply:
I guess whoever named the method did so because it can return null when the metadata isn't defined. I'll change it.